### PR TITLE
Fix csv parser

### PR DESCRIPTION
**Pull request by vext01**
@snim2 can you guve this a go?

This fixes the parser for octane by sorting the rows ahead of parsing. It looks like a big change because  block of code got de-dented.

Also make the JSON easier to inspect by hand.

OK?

### CODE REVIEW

**Comment by snim2** [comment]
OK, this works for me, will look at the code.

File: warmup/krun_results.py [hunk 1]
```diff
@@ -35,31 +35,38 @@
...
+            header = reader.next()  # Skip header row.
+            rows = iter(reader)
+            sorted_rows = sorted(rows, key=lambda l: (l[1], int(l[0])))
+
+        data_dictionary['audit']['uname'] = uname
+        expect_idx = [0]  # check we get in-order indices, first always 0
+        for row in sorted_rows:
+            # First cell contains process execution number.
+            assert int(row[0]) in expect_idx, \
+                'found gaps in process executions!'
```
Comment (snim2) [inline warmup/krun_results.py]: Can we give a little more detail here? Maybe which benchmark has gaps?

File: warmup/krun_results.py [hunk 2]
```diff
@@ -35,31 +35,38 @@
...
+            bench = row[1]
+            if row[2] == 'crash':
+                data = []
+            else:
+                data = [float(datum) for datum in row[2:]]
+            key = '%s:%s:default-%s' % (bench, vm, language)
```
Comment (snim2) [inline warmup/krun_results.py:60]: The user may have passed in a language with a `default-` prefix, so this code should check before adding it.

File: warmup/krun_results.py [hunk 3]
```diff
@@ -35,31 +35,38 @@
...
+            if key not in data_dictionary['wallclock_times']:
+                data_dictionary['wallclock_times'][key] = list()
+                data_dictionary['core_cycle_counts'][key] = list()
+                data_dictionary['aperf_counts'][key] = list()
+                data_dictionary['mperf_counts'][key] = list()
+            data_dictionary['wallclock_times'][key].append(data)
+            data_dictionary['core_cycle_counts'][key].append(None)
+            data_dictionary['aperf_counts'][key].append(None)
+            data_dictionary['mperf_counts'][key].append(None)
+            expect_idx = [0, int(row[0]) + 1]  # expect the next index, or a 0
```
Comment (snim2) [inline warmup/krun_results.py]: "expect the next index, or a 0" -> "expect the next process execution index, or a 0"

**Comment by snim2** [comment]
OK, just some minor comments.

File: warmup/krun_results.py (hunk 2 above)
Comment (vext01) [reply warmup/krun_results.py:60]: Hrm. Not 100% sure about this. Arguably this library routine should accept `variant`, not `language`. But let's not worry ourselves too much with this right now. FWIW, I did not change this logic, only indented it.

Comment (vext01) [reply warmup/krun_results.py:60]: Or, de-dented rather.

**Comment by vext01** [comment]
Fixed 2 comments, left a comment on the remaining comment.

File: warmup/krun_results.py (hunk 2 above)
Comment (snim2) [reply warmup/krun_results.py:60]: I've looked again at the help for this script and `warmup_stats` and we don't advise users to prepend the `default-`, so I think you are right, this should be left as-is.

**Comment by snim2** [comment]
OK, happy to squash?

**Comment by vext01** [comment]
splat.

**Review by snim2 — APPROVED** [review]

**Pull request merged**